src/Microsoft.AspNet.Razor.Runtime/TagHelpers/HtmlAttributeNameAttribute.cs

```diff
@@ -29,5 +31,39 @@ public HtmlAttributeNameAttribute(string name)
         /// HTML attribute name of the associated property.
         /// </summary>
         public string Name { get; }
+
+        /// <summary>
+        /// Gets or sets the prefix used to match HTML attribute names. Matching attributes are added to the
+        /// associated property (an <see cref="System.Collections.Generic.IDictionary{TKey, TValue}"/>).
+        /// </summary>
+        /// <remarks>
+        /// If non-<c>null</c> associated property must be compatible with
+        /// <see cref="System.Collections.Generic.IDictionary{TKey, TValue}"/> where <c>TKey</c> is
+        /// <see cref="string"/>.
+        /// </remarks>
+        /// <value>
+        /// If associated property is compatible with
+        /// <see cref="System.Collections.Generic.IDictionary{TKey, TValue}"/>, default value is <c>Name + "-"</c>.
+        /// Otherwise default value is <c>null</c>.
+        /// </value>
+        public string DictionaryAttributePrefix
+        {
+            get
+            {
+                return _dictionaryAttributePrefix;
+            }
+            set
+            {
+                _dictionaryAttributePrefix = value;
+                DictionaryAttributePrefixSet = true;
+            }
+        }
+
+        /// <summary>
+        /// Gets an indication whether <see cref="DictionaryAttributePrefix"/> has been set. Used to distinguish an
+        /// uninitialized <see cref="DictionaryAttributePrefix"/> value from an explicit <c>null</c> setting.
```

Comment:
What happens if the user wants to override the prefix though?

You'd have to do:

```C#
public class FooTagHelper
{
    // Need to provide a name given the ctor setup
    [HtmlAttributeName("SomeUnvaluableName", DictionaryAttributePrefix = "ANewName")]
    public IDictionary<string, string> MyDictionary { get; set; }
}
```
Which doesn't make all that much sense and is somewhat ugly.

What if we were to take an approach where we add the `-` to specify dictionary key/values, otherwise if you don't provide the `-` you can set the property directly:

```C#
[HtmlElementName("a")]
public class AnchorTagHelper
{
    public IDictionary<string, string> Params { get; set; }
}
```

Results in

```HTML
@{
    var paramDictionary = new Dictionary<string, string>
    {
        { "a", "b" }
    }
}
<!-- Would create the href appended with ?temp=SomeValue&other=OtherValue -->
<a params-temp="SomeValue" params-other="OtherValue" ...>Hello World</a>
<!-- Would create the href appended with ?a=b -->
<a params="paramDictionary" ...>Hello World</a>
<!-- Would create the href appended with ?a=b&temp=SomeValue&other=OtherValue -->
<a params-temp="SomeValue" params-other="OtherValue" params="paramDictionary" ...>Hello World</a>
```


I feel like this approach would be more flexible/intuitive.

/cc @DamianEdwards